[aliyun drive] cannot list `/`

```log
19:18:42 [DEBUG] (6) hyper_util::client::legacy::pool: pooling idle connection for ("https", openapi.alipan.com)
19:18:42 [DEBUG] (6) opendal::services::aliyun_drive::backend: backend build started: AliyunDriveBuilder { config: AliyunDriveConfig { root: None, drive_type: "", .. }, .. }
19:18:42 [DEBUG] (6) opendal::services::aliyun_drive::backend: backend use root /
19:18:42 [DEBUG] (6) opendal::services::aliyun_drive::backend: backend use drive_type Default
19:18:42 [DEBUG] (6) opendal::services: service=aliyun_drive operation=metadata -> started
19:18:42 [DEBUG] (6) opendal::services: service=aliyun_drive operation=metadata -> finished: AccessorInfo { scheme: AliyunDrive, root: "/", name: "", native_capability: { Stat | Read | Write | CreateDir | Delete | Copy | Rename | List }, full_capability: { Stat | Read | Write | CreateDir | Delete | Copy | Rename | List } }
19:18:42 [DEBUG] (6) opendal::services: service=aliyun_drive operation=list path=/ -> started
19:18:42 [DEBUG] (6) reqwest::connect: starting new connection: https://openapi.alipan.com/
19:18:42 [DEBUG] (33) hyper_util::client::legacy::connect::dns: resolving host="openapi.alipan.com"
19:18:42 [DEBUG] (6) hyper_util::client::legacy::connect::http: connecting to 39.106.251.95:443
19:18:42 [DEBUG] (7) hyper_util::client::legacy::connect::http: connected to 39.106.251.95:443
19:18:43 [DEBUG] (7) hyper_util::client::legacy::pool: pooling idle connection for ("https", openapi.alipan.com)
19:18:43 [DEBUG] (7) hyper_util::client::legacy::pool: reuse idle connection for ("https", openapi.alipan.com)
19:18:43 [DEBUG] (7) hyper_util::client::legacy::pool: pooling idle connection for ("https", openapi.alipan.com)
19:18:43 [ERROR] service=aliyun_drive operation=list path=/ -> Unexpected (permanent) at list, context: { service: aliyun_drive, path: / } => The input parameter file_path is not valid. file_path is required
19:18:43 [INFO] get_remote_storage_settings;
19:18:43 [INFO] get_remote_storage_settings_validation;
```


However, if we set "root" is the `root`, it works:

```rs
aliyun_drive_builder
                .access_token(&aliyun_drive_config.access_token)
                .root("root");
```

```logs
19:25:00 [DEBUG] (10) opendal::services: service=aliyun_drive operation=list path=/ -> start listing dir
19:25:00 [DEBUG] (10) opendal::services: service=aliyun_drive operation=list path=/ -> finished
19:25:00 [DEBUG] (10) opendal::services: service=aliyun_drive operation=list path=/ -> all entries read finished
````

--- a/core/src/services/aliyun_drive/core.rs
+++ b/core/src/services/aliyun_drive/core.rs
@@ -170,9 +170,21 @@ impl AliyunDriveCore {
         Ok((token, drive_id.clone()))
     }
 
-    pub async fn get_by_path(&self, path: &str) -> Result<Buffer> {
-        let file_path = build_rooted_abs_path(&self.root, path);
+    pub fn build_path(&self, path: &str, rooted: bool) -> String {
+        let file_path = if rooted {
+            build_rooted_abs_path(&self.root, path)
+        } else {
+            build_abs_path(&self.root, path)
+        };
         let file_path = file_path.strip_suffix('/').unwrap_or(&file_path);
+        if file_path.is_empty() {
+            return "/".to_string();
+        }
+        file_path.to_string()
+    }
+
+    pub async fn get_by_path(&self, path: &str) -> Result<Buffer> {
+        let file_path = self.build_path(path, true);
         let req = Request::post(format!(
             "{}/adrive/v1.0/openFile/get_by_path",
             self.endpoint
@@ -180,7 +192,7 @@ impl AliyunDriveCore {
         let (token, drive_id) = self.get_token_and_drive().await?;
         let body = serde_json::to_vec(&GetByPathRequest {
             drive_id: &drive_id,
-            file_path,
+            file_path: &file_path,
         })
         .map_err(new_json_serialize_error)?;
         let req = req
@@ -190,7 +202,7 @@ impl AliyunDriveCore {
     }
 
     pub async fn ensure_dir_exists(&self, path: &str) -> Result<String> {
-        let file_path = build_abs_path(&self.root, path.strip_suffix('/').unwrap_or(path));
+        let file_path = self.build_path(path, false);
         if file_path == "/" {
             return Ok("root".to_string());
         }

